src/Avalonia.Controls/TextBlock.cs

```diff
@@ -345,13 +344,55 @@ public override void Render(DrawingContext context)
             context.DrawText(Foreground, new Point(), FormattedText);
         }
 
+        protected static void AffectsFormattedText<T>(params AvaloniaProperty[] properties)
+            where T : TextBlock
+        {
+            void Handler(AvaloniaPropertyChangedEventArgs e)
+            {
+                if (e.Sender is T i)
+                {
+                    i.InvalidateFormattedText();
+                    i.InvalidateMeasure();
+                }
+            }
+
+            foreach (var property in properties)
+            {
+                property.Changed.Subscribe(Handler);
+            }
+        }
+
         /// <summary>
         /// Creates the <see cref="FormattedText"/> used to render the text.
         /// </summary>
         /// <param name="constraint">The constraint of the text.</param>
+        /// <returns>A <see cref="FormattedText"/> object.</returns>
+        protected virtual FormattedText CreateFormattedText(Size constraint)
+        {
+            var builder = new FormattedTextBuilder();
+
+            foreach (var i in Inlines)
+            {
+                i.BuildFormattedText(builder);
+            }
+
+            var result = builder.ToFormattedText();
+            result.Constraint = constraint;
+            result.TextAlignment = TextAlignment;
+            result.Typeface = new Typeface(FontFamily, FontSize, FontStyle, FontWeight);
+            result.Wrapping = TextWrapping;
+            return result;
+        }
+
+        /// <summary>
+        /// Creates a <see cref="FormattedText"/> instance using the specified text and constraint
+        /// and using the base styling properties of the <see cref="TextBlock"/>.
+        /// </summary>
+        /// <param name="constraint">The constraint of the text.</param>
         /// <param name="text">The text to format.</param>
         /// <returns>A <see cref="FormattedText"/> object.</returns>
-        protected virtual FormattedText CreateFormattedText(Size constraint, string text)
+        [Obsolete("Remove this")]
```

Comment:
Should we just remove this anyway, since the next version is going to be 0.7 and we already have several other
breaking changes?